src/orchestrator/streaming/aggregator.rs

```diff
@@ -0,0 +1,484 @@
+#![allow(dead_code)]
+use std::sync::Arc;
+
+use tokio::sync::{broadcast, mpsc, oneshot};
+use tracing::instrument;
+
+use crate::{
+    models::ClassifiedGeneratedTextStreamResult,
+    orchestrator::{
+        streaming::{Chunk, Detections},
+        Error,
+    },
+};
+
+pub type DetectorId = String;
+pub type Span = (i64, i64);
```

Comment:
Technically, no (ideally it would be `usize`). This is to avoid casts to `i64` as this is the type used for the indices in `ChunkerTokenizationStreamResult` and other places.